mne/preprocessing/eyetracking/calibration.py

```diff
@@ -0,0 +1,346 @@
+        The maximum error in degrees that occurred between the calibration points and
+        the actual gaze position.
+    points : ndarray
+        a 1D structured numpy array of shape ``(n_calibration_points,)``, which contains
+        the data for each calibration point.
+    screen_size : array-like
+        The width and height (in meters) of the screen that the eyetracking data was
+        collected  with. For example ``(.531, .298)`` for a monitor with a display area
+        of 531 x 298 mm.
+    screen_distance : float
+        The distance (in meters) from the participant's eyes to the screen.
+    screen_resolution : array-like
+        The resolution (in pixels) of the screen that the eyetracking data was
+        collected with. For example, ``(1920, 1080)`` for a 1920x1080 resolution
+        display.
+
+    Examples
+    --------
+    Below is an example of data that can be passed to the points parameter:
+    ``>>> data = [(960., 540., 0.23, 950.1,  544.1), (960., 92., 0.38, 967.8, 76. )]``
+    """
+
+    def __init__(
+        self,
+        onset=None,
+        model=None,
+        eye=None,
+        avg_error=None,
+        max_error=None,
+        points=None,
+        screen_size=None,
+        screen_distance=None,
+        screen_resolution=None,
+    ):
+        super().__init__()
+        self["onset"] = onset
+        self["model"] = model
+        self["eye"] = eye
+        self["avg_error"] = avg_error
+        self["max_error"] = max_error
+        if points is not None:
+            self.set_calibration_array(points)
+        else:
+            self["points"] = points
+        self["screen_size"] = screen_size
+        self["screen_distance"] = screen_distance
+        self["screen_resolution"] = screen_resolution
+
+    def __repr__(self):
+        """Return a summary of the Calibration object."""
+        onset = self.get("onset", "N/A")
+        model = self.get("model", "N/A")
+        eye = self.get("eye", "N/A")
+        avg_error = self.get("avg_error", "N/A")
+        max_error = self.get("max_error", "N/A")
+        screen_size = self.get("screen_size", "N/A")
+        screen_distance = self.get("screen_distance", "N/A")
+        screen_resolution = self.get("screen_resolution", "N/A")
+        return (
+            f"Calibration |\n"
+            f"  onset: {onset} seconds\n"
+            f"  model: {model}\n"
+            f"  eye: {eye}\n"
+            f"  average error: {avg_error} degrees\n"
+            f"  max error: {max_error} degrees\n"
+            f"  screen size: {screen_size} meters\n"
+            f"  screen distance: {screen_distance} meters\n"
+            f"  screen resolution: {screen_resolution} pixels\n"
+        )
+
+    def __getattr__(self, name):
+        """Allow dot indexing of dict keys."""
+        if name in self:
+            return self[name]
+        raise AttributeError(
+            f"'{self.__class__.__name__}' object has no attribute '{name}'"
+        )
```

Comment:
I'd rather not have this at all and explicitly always access by keys. But if you really want to keep this then this seems safer:
```suggestion
    def __getattr__(self, name):
        """Allow dot indexing of dict keys."""
        if name in self:
            return self[name]
        return super().__getattr__(name)
```

Reply:
No problem, in that case I won't commit suggestion then and I will remove this custom dunder method on my local.